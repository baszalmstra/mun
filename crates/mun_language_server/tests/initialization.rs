mod support;

#[async_std::test]
async fn test_server() {
    let _server = support::Project::with_fixture("").server();
}
