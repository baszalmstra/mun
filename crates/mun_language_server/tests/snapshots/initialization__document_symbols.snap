---
source: crates/mun_language_server/tests/initialization.rs
assertion_line: 50
expression: symbols

---
Some(
    Nested(
        [
            DocumentSymbol {
                name: "main",
                detail: Some(
                    "fn() -> i32",
                ),
                kind: Function,
                tags: None,
                deprecated: None,
                range: Range {
                    start: Position {
                        line: 0,
                        character: 0,
                    },
                    end: Position {
                        line: 0,
                        character: 19,
                    },
                },
                selection_range: Range {
                    start: Position {
                        line: 0,
                        character: 3,
                    },
                    end: Position {
                        line: 0,
                        character: 7,
                    },
                },
                children: None,
            },
            DocumentSymbol {
                name: "Foo",
                detail: None,
                kind: Struct,
                tags: None,
                deprecated: None,
                range: Range {
                    start: Position {
                        line: 1,
                        character: 0,
                    },
                    end: Position {
                        line: 1,
                        character: 13,
                    },
                },
                selection_range: Range {
                    start: Position {
                        line: 1,
                        character: 7,
                    },
                    end: Position {
                        line: 1,
                        character: 10,
                    },
                },
                children: None,
            },
            DocumentSymbol {
                name: "Bar",
                detail: Some(
                    "Foo",
                ),
                kind: TypeParameter,
                tags: None,
                deprecated: None,
                range: Range {
                    start: Position {
                        line: 2,
                        character: 0,
                    },
                    end: Position {
                        line: 2,
                        character: 15,
                    },
                },
                selection_range: Range {
                    start: Position {
                        line: 2,
                        character: 5,
                    },
                    end: Position {
                        line: 2,
                        character: 8,
                    },
                },
                children: None,
            },
        ],
    ),
)
